### ISSUE TITLE
NoMethodError: undefined method `[]' for <ClassName>:Class

### ISSUE THREAD

**Issue by mogman1** [jruby/jruby#6225]
I've created a test script with some specs to demonstrate this bug at https://github.com/mogman1/protobuf_error_script.

* Jruby versions: `>= 9.2.7.0`
* ActiveSupport `5.0` and `5.1`, but NOT `4.2` or `5.2` or `6.0`
* protobuf (v3.10.3)
* OSX and Ubuntu Xenial (see travis builds), probably others

The issue is that when attempting to check the value of a protobuf message attribute's truthiness by appending `?` to the end of it (e.g. `x.is_working?`), for certain versions of JRuby and ActiveSupport 5, you get something like ``` NoMethodError: undefined method `[]' for FailingProtobufMessage:Class```.  When the bug is occurring, the protobuf library is checking the class and not the instance itself when it calls `self` (see https://github.com/ruby-protobuf/protobuf/blob/v3.10.3/lib/protobuf/field/base_field.rb#L354 and error in Travis build).  This does not happen on any MRI >= 2.4 or JRuby <= 9.2.6.0.

I was able to pinpoint the exact commit where this started happening as at https://github.com/jruby/jruby/commit/3a56fcb3576fab1276a8ae51f0ea9d30a85690b9.  Everything before that my tests will all pass, and then everything from then on I get the error I mentioned.

**Comment by mogman1** [comment]
Playing around with it a little more this morning, it looks like it's susceptible to the order in which certain things are being loaded.  The protobuf gem allows you to deprecate a field and in order to trigger the bug, there must be a deprecated field but it must occur before the boolean field.  I added a test to demonstrate that.

**Comment by headius** [comment]
This is good news:

```
$ jruby -v
jruby 9.3.0.0-SNAPSHOT (2.6.5) 2020-06-09 e5d98e4865 OpenJDK 64-Bit Server VM 25.222-b10 on 1.8.0_222-b10 +jit [darwin-x86_64]

$ rake spec
/Users/headius/projects/jruby/bin/jruby -I/Users/headius/projects/jruby/lib/ruby/gems/shared/gems/rspec-support-3.9.3/lib:/Users/headius/projects/jruby/lib/ruby/gems/shared/gems/rspec-core-3.9.2/lib /Users/headius/projects/jruby/lib/ruby/gems/shared/gems/rspec-core-3.9.2/exe/rspec --pattern spec/\*\*\{,/\*/\*\*\}/\*_spec.rb
......

Finished in 0.11262 seconds (files took 5.26 seconds to load)
6 examples, 0 failures
```

@mogman1 I'm not sure what change would have solved this, but 9.3 (master) appears to run your case without errors. I'm going to close this, but it would be a big help if you could confirm with a nightly or snapshot build of JRuby from https://www.jruby.org/nightly. If you get the same error, reopen. If you get something different, open a new issue please.
